deepray/core/base_trainer.py

```diff
@@ -708,7 +709,7 @@ def run_customized_training_loop(
     # Training loop starts here.
     self.current_step = self._first_steps = self.optimizer.iterations.numpy()
 
-    self.first_batch = True
+    self.first_batch = tf.Variable(True, trainable=False, dtype=tf.bool, name='first_batch')
```

Comment:
Creating variable first_batch under init_scope would be better? Which makes sure this variable was created in eager mode to be compatible with python logical controlling code in any time, and also guarantees a safety from TF tape and other contexts.
```python
with tf.init_scope():
    # Initialization runs with eager execution enabled
    # assert tf.executing_eagerly()
    tf.Variable(True, trainable=False, dtype=tf.bool, name='first_batch')
```

Reply:
Done.